example/SimpleExample/uArchInfo.h

```diff
@@ -13,216 +13,204 @@
 #include "uArchInfoExceptions.hpp"
 #include "json.hpp"
 
+/**
+ * uArchInfo: encapsulates "static" u-arch specific information
+ */
+class uArchInfo
+{
+  public:
+    typedef std::shared_ptr<uArchInfo> PtrType;
+
+    // TODO: flesh this out
+    enum class UnitSet : uint64_t
+    {
+        AGU = 1ull << 0,
+        INT = 1ull << 1,
+        FLOAT = 1ull << 2,
+        MULTIPLY = 1ull << 3,
+        DIVIDE = 1ull << 4,
+        BRANCH = 1ull << 5,
+        LOAD = 1ull << 6,
+        STORE = 1ull << 7,
+        SYSTEM = 1ull << 8,
+        VECTOR = 1ull << 9,
+    };
+
+    // BEGIN: Stuff imported from the old StaticInstructionData object
+    enum RegFile
+    {
+        INTEGER,
+        FLOAT,
+        INVALID,
+        N_REGFILES = INVALID
+    };
+
+    static inline const char* const regfile_names[] = {"integer", "float"};
+
+    // TODO: resolve this against the UnitSet
+    enum IssueTarget : std::uint16_t
+    {
+        IEX,
+        FEX,
+        BR,
+        LSU,
+        ROB, // Instructions that go right to retire
+        N_ISSUE_TARGETS
+    };
+
+    static constexpr uint32_t MAX_ARCH_REGS = 5;
+    // END: Stuff imported from the old StaticInstructionData object
+
+  private:
+    // Map unit names to unit ID's
+    static inline std::map<std::string, UnitSet> umap_ = {
+        {"agu", UnitSet::AGU},       {"int", UnitSet::INT},     {"float", UnitSet::FLOAT},
+        {"mul", UnitSet::MULTIPLY},  {"div", UnitSet::DIVIDE},  {"branch", UnitSet::BRANCH},
+        {"load", UnitSet::LOAD},     {"store", UnitSet::STORE}, {"system", UnitSet::SYSTEM},
```

Comment:
Wonder what format flag we can employ that will put this code back to the original formatting...

Reply:
https://github.com/sparcians/mavis/pull/46 has an attempt at fixing this.